cli/cli/commands/files/inspect/inspect.go

```diff
@@ -188,3 +196,48 @@ func humanReadableSize(size uint64) string {
 	}
 	return fmt.Sprintf("%3.1f%c", fSize, sizeSuffix[suffixIdx])
 }
+
+func getCompletionFunc(enclaveArgKey string, artifactArgKey string) func(ctx context.Context, _ *flags.ParsedFlags, previousArgs *args.ParsedArgs) ([]string, error) {
+	return func(ctx context.Context, _ *flags.ParsedFlags, previousArgs *args.ParsedArgs) ([]string, error) {
+		kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+		if err != nil {
+			return nil, stacktrace.Propagate(
+				err,
+				"An error occurred connecting to the Kurtosis engine for retrieving the names for tab completion",
+			)
+		}
+
+		enclaveIdentifier, err := previousArgs.GetNonGreedyArg(enclaveArgKey)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred getting the enclave ID using key '%v'", enclaveArgKey)
+		}
+		enclave, err := kurtosisCtx.GetEnclaveContext(ctx, enclaveIdentifier)
+		if err != nil {
+			return nil, stacktrace.Propagate(
+				err,
+				"An error occurred getting the enclave identifiers",
+			)
+		}
+
+		artifactIdentifier, err := previousArgs.GetNonGreedyArg(artifactArgKey)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred getting the artifact ID using key '%v'", artifactArgKey)
+		}
+		fileArtifactContents, err := enclave.InspectFilesArtifact(ctx, services.FileArtifactName(artifactIdentifier))
+		if err != nil {
+			return nil, stacktrace.Propagate(
+				err,
+				"An error occurred getting the file artifacts",
+			)
+		}
+		fileArtifactContentPaths := []string{}
+		for _, fileArtifactDescription := range fileArtifactContents.GetFileDescriptions() {
+			fileArtifactContentPath := fileArtifactDescription.GetPath()
+			if fileArtifactDescription.TextPreview != nil {
```

Comment:
why is completion only for those without text preview?

Reply:
Printing file contents for files we have no contents yields no meaningful result. I can add a follow up to print something like "no file contents exist"